cmake/macros/SundialsAddLibrary.cmake

```diff
@@ -239,25 +247,22 @@ macro(sundials_add_library target)
       # set target name
       set(_actual_target_name ${target}${_lib_suffix})
 
-      add_library(${_actual_target_name} ${_libtype} $<TARGET_OBJECTS:${obj_target}>)
-
-      set_target_properties(${_actual_target_name} PROPERTIES FOLDER "src")
-
-      # add any object library dependencies
+      set(_object_sources $<TARGET_OBJECTS:${obj_target}>)
       if(sundials_add_library_OBJECT_LIBRARIES)
         if(${_libtype} MATCHES "STATIC")
           append_static_suffix(sundials_add_library_OBJECT_LIBRARIES _all_objs)
         else()
           set(_all_objs ${sundials_add_library_OBJECT_LIBRARIES})
         endif()
         foreach(_tmp ${_all_objs})
-          # We use target_sources since target_link_libraries does not work
```

Comment:
Ill note that we cannot actually use `target_link_libraries` even though we have newer CMake now because we dont actually export these object targets.